### ISSUE TITLE
Faces end up looking more caucasian 

### ISSUE THREAD

**Issue by buckwheaton** [Bing-su/adetailer#92]
Is there a way to feed your prompt, or some elements, into adetailer automatically?  I use a nationality wildcard and non-white folks always end up looking like second or third generation mixed after the adetailer run.  But I can't tell in advance of a gen whether someone will be Japanese or Brazilian or whatever so I can't prestage anything in the positive prompt.  Lowering the denoise level might work, I guess, but is there another solution?  Using mediapipe_full with the controlnet inpainting model.

**Comment by Bing-su** [comment]
If you leave the ad_prompt blank, it will copy the input prompt (with wildcards and dynamic prompts applied).

**Comment by buckwheaton** [comment]
> If you leave the ad_prompt blank, it will copy the input prompt (with wildcards and dynamic prompts applied).

Thanks - that makes sense.  Any idea why folks would get more caucasian looking if the prompt with nationality is being fed through to AD?

**Comment by Bing-su** [comment]
I have no idea about that. sorry.

**Comment by marhensa** [comment]
I see similar result, faces turns into Caucasian descendants, but only when applying Hires fix. aDetailer is processing and running after Hires fix process, and this I think that makes face is Caucasian-like.

tried with blank (so it's using initial prompt), or with copy paste the prompt into aDetailer positive prompt, it still turns Caucasian-like.

BUT I found out that be a fix for me is, it's counterintuitive, I use aDetailer to fix face in initial image generation, then I sent it to Img2img to simulate Hires fix, the face result is much better (not Caucasian-like), similar to what aDetailer generates initially.

tl;dr
1. applying aDetailer without Hires fix, face is correct (not Caucasian-like), but the resolution is obviously small
2. applying aDetailer with Hires fix, the aDetailer will run AFTER Hires fix is done, face is Caucasian-like (I don't know why, could be model, could be anything else beside extension)
3. applying aDetailer in initial image generation, then use Img2img to simulate Hires fix, face are generated better, resembles result from aDetailer before upscaling

So can we get options to run aDetailer BEFORE Hires fix? not after.

**Comment by Bing-su** [comment]
It seems simple to do, but the current implementation doesn't allow ADetailer to work before hires, sorry.

**Comment by andupotorac** [comment]
> I see similar result, faces turns into Caucasian descendants, but only when applying Hires fix. aDetailer is processing and running after Hires fix process, and this I think that makes face is Caucasian-like.
> 
> tried with blank (so it's using initial prompt), or with copy paste the prompt into aDetailer positive prompt, it still turns Caucasian-like.
> 
> BUT I found out that be a fix for me is, it's counterintuitive, I use aDetailer to fix face in initial image generation, then I sent it to Img2img to simulate Hires fix, the face result is much better (not Caucasian-like), similar to what aDetailer generates initially.
> 
> tl;dr
> 
> 1. applying aDetailer without Hires fix, face is correct (not Caucasian-like), but the resolution is obviously small
> 2. applying aDetailer with Hires fix, the aDetailer will run AFTER Hires fix is done, face is Caucasian-like (I don't know why, could be model, could be anything else beside extension)
> 3. applying aDetailer in initial image generation, then use Img2img to simulate Hires fix, face are generated better, resembles result from aDetailer before upscaling
> 
> So can we get options to run aDetailer BEFORE Hires fix? not after.

Try using this TI, let me know if it works. 
1. https://civitai.com/models/50755/style-asian-less
